src/js/utils/FormUtil.js

```diff
@@ -0,0 +1,209 @@
+import Util from './Util';
+
+function getProp(key) {
+  return key.split('[')[0];
+}
+
+function getPropIndex(key) {
+  let value = key.split('[');
+  return parseInt(value[1].split(']')[0])
+}
+
+function getPropKey(key) {
+  return key.split('.')[1];
+}
+
+function isNotMultipleProp(key) {
+  return !key.includes('[') || !key.includes(']');
+}
+
+function containsMultipleProp(prop, id, fieldColumn) {
+  return !!(
+    fieldColumn &&
+    fieldColumn.name &&
+    fieldColumn.name.includes(`${prop}[${id}]`)
+  );
+}
+
+function getNewDefinition(prop, propID, parentDefinition, itemDefinition, getRemoveButton, model) {
+  let newDefinition = FormUtil.getMultipleFieldDefinition(
+    prop,
+    propID,
+    itemDefinition,
+    model
+  );
+
+  // Add remove button.
+  if (getRemoveButton) {
+    newDefinition.push(
+      getRemoveButton(parentDefinition, prop, propID)
+    );
+  }
+
+  return newDefinition;
+}
+
+const FormUtil = {
+  /**
+   * @return {Array} parentDefinition the modified definition.
+   * @param {Object} options Object containing properties for
+   * transformingDefinition. The following properties are:
```

Comment:
Interesting organization of JDOC, could we keep [it consistent](https://github.com/dcos/dcos-ui/blob/master/src/js/utils/TableUtil.js#L59) with others?